hpccm/templates/ConfigureCMake.py

```diff
@@ -0,0 +1,139 @@
+                             "configure_step!")
+        if not os.path.isabs(directory):
+            raise ValueError("directory given in ConfigureCMake configure_step"
+                             "has to be an absolute path!")
+
+        if not os.path.isabs(build_directory):
+            build_directory = os.path.join(directory, build_directory)
+
+        self.__build_directory = build_directory
+
+        change_directory = "mkdir -p {0} && cd {0} && ".format(
+            self.__build_directory)
+
+        prefix = []
+        if toolchain:
+            if toolchain.CC and self.toolchain_control.get('CC'):
+                prefix.append('CC={}'.format(toolchain.CC))
+
+            if toolchain.CFLAGS:
+                prefix.append('CFLAGS={}'.format(shlex_quote(
+                    toolchain.CFLAGS)))
+
+            if toolchain.CPPFLAGS:
+                prefix.append('CPPFLAGS={}'.format(shlex_quote(
+                    toolchain.CPPFLAGS)))
+
+            if toolchain.CXX and self.toolchain_control.get('CXX'):
+                prefix.append('CXX={}'.format(toolchain.CXX))
+
+            if toolchain.CXXFLAGS:
+                prefix.append('CXXFLAGS={}'.format(shlex_quote(
+                    toolchain.CXXFLAGS)))
+
+            if toolchain.F77 and self.toolchain_control.get('F77'):
+                prefix.append('F77={}'.format(toolchain.F77))
+
+            if toolchain.F90 and self.toolchain_control.get('F90'):
+                prefix.append('F90={}'.format(toolchain.F90))
+
+            if toolchain.FC and self.toolchain_control.get('FC'):
+                prefix.append('FC={}'.format(toolchain.FC))
+
+            if toolchain.FCFLAGS:
+                prefix.append('FCFLAGS={}'.format(shlex_quote(
+                    toolchain.FCFLAGS)))
+
+            if toolchain.FFLAGS:
+                prefix.append('FFLAGS={}'.format(shlex_quote(
+                    toolchain.FFLAGS)))
+
+            if toolchain.FLIBS:
+                prefix.append('FLIBS={}'.format(shlex_quote(
+                    toolchain.FLIBS)))
+
+            if toolchain.LD_LIBRARY_PATH:
+                prefix.append('LD_LIBRARY_PATH={}'.format(shlex_quote(
+                    toolchain.LD_LIBRARY_PATH)))
+
+            if toolchain.LDFLAGS:
+                prefix.append('LDFLAGS={}'.format(shlex_quote(
+                    toolchain.LDFLAGS)))
+
+            if toolchain.LIBS:
+                prefix.append('LIBS={}'.format(shlex_quote(
+                    toolchain.LIBS)))
+
+        configure_prefix = ' '.join(prefix)
+        if configure_prefix != '':
+            configure_prefix += ' '
+        
+        configure_opts = ''
+        if opts:
+            configure_opts = ' '.join(opts)
+            configure_opts += ' '
+
+        cmd = '{0}{1}cmake {2}{3}'.format(
+            change_directory, configure_prefix, configure_opts, directory)
+
+        return cmd.strip()  # trim whitespace
+
+    def install_step(self):
```

Comment:
Should there also be a `check_step`, with a similar implementation as `install_step`?

Reply:
I was not even sure to include the `install_step` because it is not necessary (just use `build_step(target='install')`). At least the `install`-target is a predefined one.

There is no such predefined target for running tests. Often people use `test` as the target name but this has to be explicitly defined by each CMake project.

So what do you think should I add a `check_step` or should I even remove the `install_step`?

Reply:
OK with removing `install_step`.